### ISSUE TITLE
Add path-only normlaization

### ISSUE THREAD

**Issue by lptr** [gradle/gradle#10416]
Probably not a huge feature, but if we are in the business of adding more normalizations, this might be nice:

When a task only uses the _names_ or _relative paths_ of its input files (e.g. `CreateStartScripts.relativeClasspath`) we still want to track build dependencies. This can be tricky involving an `@Input List<String>` property, an `@internal FileColletion` one, and no `@InputFile`s or `@Classpath`. This is confusing, and we could circumvent it by adding a specific normalization for file paths instead.

---
cc: @gradle/build-cache

**Comment by wolfs** [comment]
`Property` types now track build dependencies, so `@Input` is perfectly fine, isn't it?

**Comment by lptr** [comment]
That's true. However, you still need an `@Internal ConfigurableFileCollection` property, or the user has to do the mapping from files to paths, which is error-prone.

```groovy
@NoContent
@PathSensitive(RELATIVE)
ConfigurableFileCollection classpath
```

The above could be easier to comprehend.

**Comment by wolfs** [comment]
This is something we want to do.
